script/makeraw_unix.go

```diff
@@ -0,0 +1,43 @@
+//go:build darwin || linux
+
+package script
+
+import (
+	"runtime"
+
+	"golang.org/x/sys/unix"
+)
+
+// MakeRaw sets the terminal to raw mode, but with interrupt signals enabled.
+func MakeRaw(fd int) (restore func(), err error) {
+	var ioctlReadTermios, ioctlWriteTermios uint
+	if runtime.GOOS == "darwin" {
+		ioctlReadTermios = 0x40487413  // unix.TIOCGETA
```

Comment:
x/term does this by having `unix_other.go` and `unix_bsd.go` which use the different constants. If preferred I can do that, but this IMO works as well.